File: src/SEUnitButtonPartClass.cs
// Decompiled with JetBrains decompiler
// Type: WindowsApplication1.SEUnitButtonPartClass
// Assembly: WindowsApplication1, Version=1.0.8020.28903, Culture=neutral, PublicKeyToken=null
// MVID: F52869E5-0850-48AD-BBBE-68E7A4900AFE
// Assembly location: C:\Program Files (x86)\Steam\steamapps\common\Shadow Empire\ShadowEmpire.exe

using Microsoft.VisualBasic;
using System.Drawing;

namespace WindowsApplication1
{
  public class SEUnitButtonPartClass : SubPartClass
  {
    private int iconSlotNr;
    private string dataString;
    private string dataString2;
    private string description;
    private bool active;
    private int overruleR;
    private int overruleG;
    private int overruleB;

    public override void SubDispose()
    {
    }

    public SEUnitButtonPartClass(
      int tIconSlotNr,
      string tDataString,
      string tDataString2,
      string tDescript,
      bool tactive,
      int tOverruleR = -1,
      int tOverruleG = -1,
      int tOverruleB = -1)
      : base(44, 97)
    {
      this.iconSlotNr = tIconSlotNr;
      this.Descript = tDescript;
      this.dataString = tDataString;
      this.dataString2 = tDataString2;
      this.active = tactive;
      this.overruleG = tOverruleG;
      this.overruleB = tOverruleB;
      this.overruleR = tOverruleR;
    }

    public override Bitmap Paint()
    {
      Graphics objgraphics = Graphics.FromImage((Image) this.OwnBitmap);
      Bitmap bitmap;
      if (!this.active)
      {
        ref Graphics local1 = ref objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_VARBOX5);
        ref Bitmap local2 = ref bitmap;
        DrawMod.DrawSimple(ref local1, ref local2, 0, 0);
      }
      if (this.active)
      {
        ref Graphics local3 = ref objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_VARBOX5HIGH);
        ref Bitmap local4 = ref bitmap;
        DrawMod.DrawSimple(ref local3, ref local4, 0, 0);
      }
      Rectangle rectangle1;
      Rectangle rectangle2;
      if (!this.active)
      {
        ref Graphics local5 = ref objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_ICONS);
        ref Bitmap local6 = ref bitmap;
        rectangle1 = new Rectangle(this.iconSlotNr * 42, 0, 42, 32);
        Rectangle srcrect = rectangle1;
        rectangle2 = new Rectangle(2, 10, 42, 32);
        Rectangle destrect = rectangle2;
        DrawMod.DrawSimplePart2(ref local5, ref local6, srcrect, destrect);
      }
      if (this.active)
      {
        ref Graphics local7 = ref objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_ICONS);
        ref Bitmap local8 = ref bitmap;
        rectangle2 = new Rectangle(this.iconSlotNr * 42, 32, 42, 32);
        Rectangle srcrect = rectangle2;
        rectangle1 = new Rectangle(2, 10, 42, 32);
        Rectangle destrect = rectangle1;
        DrawMod.DrawSimplePart2(ref local7, ref local8, srcrect, destrect);
      }
      if (this.overruleB > -1)
      {
        if (this.dataString2.Length > 0)
        {
          if (!this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 51, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
          if (this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 51, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
          if (!this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString2, DrawMod.TGame.MarcFont16, 22, 65, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
          if (this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString2, DrawMod.TGame.MarcFont16, 22, 65, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
        }
        else
        {
          if (!this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 58, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
          if (this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 58, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
        }
      }
      else if (this.dataString2.Length > 0)
      {
        if (!this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 51, DrawMod.TGame.seColGray);
        if (this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 51, DrawMod.TGame.seColWhite);
        if (!this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString2, DrawMod.TGame.MarcFont16, 22, 65, DrawMod.TGame.seColGray);
        if (this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString2, DrawMod.TGame.MarcFont16, 22, 65, DrawMod.TGame.seColWhite);
      }
      else
      {
        if (!this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 58, DrawMod.TGame.seColGray);
        if (this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 58, DrawMod.TGame.seColWhite);
      }
      if (!Information.IsNothing((object) objgraphics))
        objgraphics.Dispose();
      return this.OwnBitmap;
    }

    public override Bitmap PaintOverlay()
    {
      Graphics objgraphics = Graphics.FromImage((Image) this.OwnBitmap);
      Bitmap bitmap;
      if (!this.active)
      {
        ref Graphics local1 = ref objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_VARBOX5);
        ref Bitmap local2 = ref bitmap;
        DrawMod.DrawSimple(ref local1, ref local2, 0, 0);
        ref Graphics local3 = ref objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_VARBOX5HIGH);
        ref Bitmap local4 = ref bitmap;
        DrawMod.Draw(ref local3, ref local4, 0, 0, 0.0f, 0.0f, 0.0f, 0.2f);
      }
      if (this.active)
      {
        ref Graphics local5 = ref objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_VARBOX5HIGH);
        ref Bitmap local6 = ref bitmap;
        DrawMod.DrawSimple(ref local5, ref local6, 0, 0);
      }
      Rectangle rectangle1;
      Rectangle rectangle2;
      if (!this.active)
      {
        ref Graphics local7 = ref objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_ICONS);
        ref Bitmap local8 = ref bitmap;
        rectangle1 = new Rectangle(this.iconSlotNr * 42, 0, 42, 32);
        Rectangle srcrect = rectangle1;
        rectangle2 = new Rectangle(2, 10, 42, 32);
        Rectangle destrect = rectangle2;
        DrawMod.DrawSimplePart2(ref local7, ref local8, srcrect, destrect);
      }
      if (this.active)
      {
        ref Graphics local9 = ref objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_ICONS);
        ref Bitmap local10 = ref bitmap;
        rectangle2 = new Rectangle(this.iconSlotNr * 42, 32, 42, 32);
        Rectangle srcrect = rectangle2;
        rectangle1 = new Rectangle(2, 10, 42, 32);
        Rectangle destrect = rectangle1;
        DrawMod.DrawSimplePart2(ref local9, ref local10, srcrect, destrect);
      }
      if (this.overruleB > -1)
      {
        if (this.dataString2.Length > 0)
        {
          if (!this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 51, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
          if (this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 51, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
          if (!this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString2, DrawMod.TGame.MarcFont16, 22, 65, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
          if (this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString2, DrawMod.TGame.MarcFont16, 22, 65, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
        }
        else
        {
          if (!this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 58, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
          if (this.active)
            DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 58, Color.FromArgb((int) byte.MaxValue, this.overruleR, this.overruleG, this.overruleB));
        }
      }
      else if (this.dataString2.Length > 0)
      {
        if (!this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 51, DrawMod.TGame.seColGray);
        if (this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 51, DrawMod.TGame.seColWhite);
        if (!this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString2, DrawMod.TGame.MarcFont16, 22, 65, DrawMod.TGame.seColGray);
        if (this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString2, DrawMod.TGame.MarcFont16, 22, 65, DrawMod.TGame.seColWhite);
      }
      else
      {
        if (!this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 58, DrawMod.TGame.seColGray);
        if (this.active)
          DrawMod.DrawTextColouredConsoleCenter(ref objgraphics, this.dataString, DrawMod.TGame.MarcFont16, 22, 58, DrawMod.TGame.seColWhite);
      }
      if (!Information.IsNothing((object) objgraphics))
        objgraphics.Dispose();
      return this.OwnBitmap;
    }
  }
}

File: src/SEUnitButtonPartClass.rs
// Decompiled with JetBrains decompiler
// Type: WindowsApplication1.SEUnitButtonPartClass
// Assembly: WindowsApplication1, Version=1.0.8020.28903, Culture=neutral, PublicKeyToken=null
// MVID: F52869E5-0850-48AD-BBBE-68E7A4900AFE
// Assembly location: C:\Program Files (x86)\Steam\steamapps\common\Shadow Empire\ShadowEmpire.exe

// usingMicrosoft.VisualBasic;
// usingSystem.Drawing;

namespace WindowsApplication1
{
  pub class SEUnitButtonPartClass : SubPartClass
  {
     iconSlotNr: i32;
     dataString: String;
     dataString2: String;
     description: String;
     bool active;
     overruleR: i32;
     overruleG: i32;
     overruleB: i32;

    pub fn SubDispose()
    {
    }

    pub SEUnitButtonPartClass(
      tIconSlotNr: i32,
      tDataString: String,
      tDataString2: String,
      tDescript: String,
      bool tactive,
      let mut tOverruleR: i32 = -1,
      let mut tOverruleG: i32 = -1,
      let mut tOverruleB: i32 = -1)
      : base(44, 97)
    {
      self.iconSlotNr = tIconSlotNr;
      self.Descript = tDescript;
      self.dataString = tDataString;
      self.dataString2 = tDataString2;
      self.active = tactive;
      self.overruleG = tOverruleG;
      self.overruleB = tOverruleB;
      self.overruleR = tOverruleR;
    }

    pub Paint: Bitmap()
    {
      Graphics objgraphics = Graphics.FromImage((Image) self.OwnBitmap);
      bitmap: Bitmap;
      if (!self.active)
      {
         let mut local1: &Graphics = &objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_VARBOX5);
         let mut local2: &Bitmap = &bitmap;
        DrawMod.DrawSimple( local1,  local2, 0, 0);
      }
      if (self.active)
      {
         let mut local3: &Graphics = &objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_VARBOX5HIGH);
         let mut local4: &Bitmap = &bitmap;
        DrawMod.DrawSimple( local3,  local4, 0, 0);
      }
      Rectangle rectangle1;
      Rectangle rectangle2;
      if (!self.active)
      {
         let mut local5: &Graphics = &objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_ICONS);
         let mut local6: &Bitmap = &bitmap;
        rectangle1 = Rectangle::new(self.iconSlotNr * 42, 0, 42, 32);
        let mut srcrect: &Rectangle = &rectangle1
        rectangle2 = Rectangle::new(2, 10, 42, 32);
        let mut destrect: &Rectangle = &rectangle2
        DrawMod.DrawSimplePart2( local5,  local6, srcrect, destrect);
      }
      if (self.active)
      {
         let mut local7: &Graphics = &objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_ICONS);
         let mut local8: &Bitmap = &bitmap;
        rectangle2 = Rectangle::new(self.iconSlotNr * 42, 32, 42, 32);
        let mut srcrect: &Rectangle = &rectangle2
        rectangle1 = Rectangle::new(2, 10, 42, 32);
        let mut destrect: &Rectangle = &rectangle1
        DrawMod.DrawSimplePart2( local7,  local8, srcrect, destrect);
      }
      if (self.overruleB > -1)
      {
        if (self.dataString2.Length > 0)
        {
          if (!self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 51, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
          if (self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 51, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
          if (!self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString2, DrawMod.TGame.MarcFont16, 22, 65, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
          if (self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString2, DrawMod.TGame.MarcFont16, 22, 65, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
        }
        else
        {
          if (!self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 58, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
          if (self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 58, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
        }
      }
      else if (self.dataString2.Length > 0)
      {
        if (!self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 51, DrawMod.TGame.seColGray);
        if (self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 51, DrawMod.TGame.seColWhite);
        if (!self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString2, DrawMod.TGame.MarcFont16, 22, 65, DrawMod.TGame.seColGray);
        if (self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString2, DrawMod.TGame.MarcFont16, 22, 65, DrawMod.TGame.seColWhite);
      }
      else
      {
        if (!self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 58, DrawMod.TGame.seColGray);
        if (self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 58, DrawMod.TGame.seColWhite);
      }
      if (!Information.IsNothing( objgraphics))
        objgraphics.Dispose();
      return self.OwnBitmap;
    }

    pub PaintOverlay: Bitmap()
    {
      Graphics objgraphics = Graphics.FromImage((Image) self.OwnBitmap);
      bitmap: Bitmap;
      if (!self.active)
      {
         let mut local1: &Graphics = &objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_VARBOX5);
         let mut local2: &Bitmap = &bitmap;
        DrawMod.DrawSimple( local1,  local2, 0, 0);
         let mut local3: &Graphics = &objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_VARBOX5HIGH);
         let mut local4: &Bitmap = &bitmap;
        DrawMod.Draw( local3,  local4, 0, 0, 0.0f, 0.0f, 0.0f, 0.2f);
      }
      if (self.active)
      {
         let mut local5: &Graphics = &objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_VARBOX5HIGH);
         let mut local6: &Bitmap = &bitmap;
        DrawMod.DrawSimple( local5,  local6, 0, 0);
      }
      Rectangle rectangle1;
      Rectangle rectangle2;
      if (!self.active)
      {
         let mut local7: &Graphics = &objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_ICONS);
         let mut local8: &Bitmap = &bitmap;
        rectangle1 = Rectangle::new(self.iconSlotNr * 42, 0, 42, 32);
        let mut srcrect: &Rectangle = &rectangle1
        rectangle2 = Rectangle::new(2, 10, 42, 32);
        let mut destrect: &Rectangle = &rectangle2
        DrawMod.DrawSimplePart2( local7,  local8, srcrect, destrect);
      }
      if (self.active)
      {
         let mut local9: &Graphics = &objgraphics;
        bitmap = BitmapStore.GetBitmap(DrawMod.TGame.SE1_ICONS);
         let mut local10: &Bitmap = &bitmap;
        rectangle2 = Rectangle::new(self.iconSlotNr * 42, 32, 42, 32);
        let mut srcrect: &Rectangle = &rectangle2
        rectangle1 = Rectangle::new(2, 10, 42, 32);
        let mut destrect: &Rectangle = &rectangle1
        DrawMod.DrawSimplePart2( local9,  local10, srcrect, destrect);
      }
      if (self.overruleB > -1)
      {
        if (self.dataString2.Length > 0)
        {
          if (!self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 51, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
          if (self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 51, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
          if (!self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString2, DrawMod.TGame.MarcFont16, 22, 65, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
          if (self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString2, DrawMod.TGame.MarcFont16, 22, 65, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
        }
        else
        {
          if (!self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 58, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
          if (self.active)
            DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 58, Color.FromArgb( byte.MaxValue, self.overruleR, self.overruleG, self.overruleB));
        }
      }
      else if (self.dataString2.Length > 0)
      {
        if (!self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 51, DrawMod.TGame.seColGray);
        if (self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 51, DrawMod.TGame.seColWhite);
        if (!self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString2, DrawMod.TGame.MarcFont16, 22, 65, DrawMod.TGame.seColGray);
        if (self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString2, DrawMod.TGame.MarcFont16, 22, 65, DrawMod.TGame.seColWhite);
      }
      else
      {
        if (!self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 58, DrawMod.TGame.seColGray);
        if (self.active)
          DrawMod.DrawTextColouredConsoleCenter( objgraphics, self.dataString, DrawMod.TGame.MarcFont16, 22, 58, DrawMod.TGame.seColWhite);
      }
      if (!Information.IsNothing( objgraphics))
        objgraphics.Dispose();
      return self.OwnBitmap;
    }
  }
}
